### ISSUE TITLE
Conv functions not fully compatible with keras when stride is used

### ISSUE THREAD

**Issue by f12markovic** [ARM-software/CMSIS_5#831]
Since keras uses tensorflow convention, the conv kernel shapes are reversed `(kernel_height, kernel_width, input_channels, output_channels)`. But also, calculating convolution is done in reverse, which means it starts from the lower right corner. Having strides can potentially calculate convolution on different pixels (for example if height/width is even and `stride=2` is used). Though this isn't a huge problem, since the result of these convolutions should be similar, the issue can arrive at the edge cases, when padding is applied. For example, if we have an image with shape `(8, 16)` and with all pixels having the same value, the conv output with `stride=2` and zero-padding will look like this (using real-case values):

- CMSIS output (conv is done starting from the upper left corner => padding is applied only to top row and left column):
```
    -1.2912    -1.9512    -1.9512    -1.9512    -1.9512    -1.9512    -1.9512    -1.9512 
    -0.0380    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759
    -0.0380    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759
    -0.0380    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759
```

- Keras output (conv is done starting from the lower right corner => padding is applied only to bottom row and right column):
```
    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -0.5210
    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -0.5210    
    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -1.1759    -0.5210    
     0.7810     0.7810     0.7810     0.7810     0.7810     0.7810     0.7810     0.2834     
```

I guess this wouldn't matter for bigger images, but in my case, the output of the last conv layer led to really weird results. 

My solution was to add offset to the starting position of the kernel, where 
`offsetX = (inX - 1) % strideX`  and 
`offsetY = (inY - 1) % strideY`.

**Comment by felix-johnny** [comment]
@f12markovic Could you give some more info on the setup that you are using? Are you using the legacy or Pre-TFLu functions (https://github.com/ARM-software/CMSIS_5/tree/develop/CMSIS/NN#legacy-vs-tfl-micro-compliant-apis) . Reason I ask is, the layout for filters in convolution is [output channel, height, width, input channel] for the TFLu compatible ones. 
  If it is the legacy ones, was the model converted into a CMSIS-NN compatible one as specified in the reference of the link mentioned above?
